libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -433,15 +433,27 @@ bool AP_Arming::gps_checks(bool report)
     const AP_GPS &gps = AP::gps();
     if ((checks_to_perform & ARMING_CHECK_ALL) || (checks_to_perform & ARMING_CHECK_GPS)) {
 
+        bool gps1_status_fail = false;
+        bool gps2_status_fail = false;
+
         //GPS OK?
         if (!AP::ahrs().home_is_set() ||
             gps.status() < AP_GPS::GPS_OK_FIX_3D) {
             check_failed(ARMING_CHECK_GPS, report, "Bad GPS Position");
             return false;
         }
 
-        if (gps.status() < gps.status_arm_min()) {
-            check_failed(ARMING_CHECK_GPS, report, "Waiting for enhanced GPS fix");
+        if (gps.status(0) < gps.status_arm_min()) {
```

Comment:
If one of the gps modules fail, will this get function return safely?

Reply:
Yes, It returns a value of 0 if that GPS instance is not active (due to failure or other reasons such as wrong configuration etc).